### PR TITLE
Add config apply rest endpoint

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Added 2 new endpoints to the Agent REST API:

- `PUT /nginx/config/`
- `GET /nginx/config/status?correlation_id=...`

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

**Review by aphralG — APPROVED** [review]

File: src/plugins/agent_api_test.go [hunk 1]
```diff
@@ -30,8 +35,74 @@ import (
 	"github.com/nginx/agent/v2/src/core/config"
 	tutils "github.com/nginx/agent/v2/test/utils"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 )
 
+const (
```
Comment (oliveromahony) [inline src/plugins/agent_api_test.go:42]: we could provide this in test utils?

**Comment by oliveromahony** [comment]
can you rebase?

File: src/plugins/agent_api_test.go (hunk 1 above)
Comment (dhurley) [reply src/plugins/agent_api_test.go]: moved to test utils

**Comment by dhurley** [comment]
> can you rebase?

Yeah merged the latest from main into the branch

**Review by Dean-Coakley — APPROVED** [review]

File: src/plugins/agent_api.go [hunk 2]
```diff
@@ -136,7 +136,9 @@ func (a *AgentAPI) Process(message *core.Message) {
 	case core.NginxConfigValidationPending, core.NginxConfigApplyFailed, core.NginxConfigApplySucceeded:
 		switch response := message.Data().(type) {
 		case *proto.AgentActivityStatus:
-			a.nginxHandler.configResponseStatus = response.GetNginxConfigStatus()
+			log.Error(response)
```
Comment (Dean-Coakley) [inline src/plugins/agent_api.go:139]:
```suggestion
```

**Review by Dean-Coakley — APPROVED** [review]

File: src/plugins/agent_api.go (hunk 2 above)
Comment (dhurley) [reply src/plugins/agent_api.go]: updated

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**